### PR TITLE
TDL-9809: `forced-replication-method` missing from metadata for some streams and TDL-9872: replication keys are not specified as expected in discoverable metadata	

### PR DESCRIPTION
**Pull request by savan-chovatiya**
# Description of change
TDL-9809: forced-replication-method missing from metadata for some streams
- Added replication method under `forced-replication-method`.
- Updated the discovery test to verify the scenario.

TDL-9872: replication keys are not specified as expected in discoverable metadata
- Added replication keys under `valid-replication-keys`.
- Updated the discovery test to verify the scenario.

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_facebook/__init__.py [hunk 1]
```diff
@@ -284,6 +285,7 @@ class Ads(IncrementalStream):
     '''
 
     key_properties = ['id', 'updated_time']
+    replication_method = 'INCREMENTAL'
```
Comment (dbshah1212) [inline tap_facebook/__init__.py:288]: Is it possible to define it in the `IncrementalStream` class instead of defining it in all classes?

File: tap_facebook/__init__.py [hunk 2]
```diff
@@ -268,6 +268,7 @@ def sync_batches(self, stream_objects):
         api_batch.execute()
 
     key_properties = ['id']
+    replication_method = 'FULL_TABLE'
```
Comment (dbshah1212) [inline tap_facebook/__init__.py:271]: Is it possible to define it in the `Stream` class?

File: tap_facebook/__init__.py (hunk 1 above)
Comment (savan-chovatiya) [reply tap_facebook/__init__.py]: Defined incremental replication key in  `IncrementalStream` class and removed from extended class. Leads and AdsInsights are incremental but extended from `Stream` so keep it as it is.

File: tap_facebook/__init__.py (hunk 2 above)
Comment (savan-chovatiya) [reply tap_facebook/__init__.py]: Yes, Defined the FULL_TABLE replication key in the `Stream` class removed from the extended class.

**Review by dbshah1212 — APPROVED** [review]

File: tests/test_facebook_discovery.py [hunk 3]
```diff
@@ -84,40 +84,26 @@ def test_run(self):
...
-                    # verify that if there is a replication key we are doing INCREMENTAL otherwise FULL
-                    if actual_replication_keys:
-                        self.assertEqual(self.INCREMENTAL, actual_replication_method)
-                    else:
-                        self.assertEqual(self.FULL_TABLE, actual_replication_method)
+                # verify the replication method matches our expectations
+                self.assertEqual(
+                    expected_replication_method, actual_replication_method
+                )
+
```
Comment (karanpanchal-crest) [inline tests/test_facebook_discovery.py:101]: @savan-chovatiya What is the difference between the check you have done at line 98 and line 103?
In both the checks you are checking the value of `actual_replication_method`

Comment (savan-chovatiya) [reply tests/test_facebook_discovery.py:101]: We have checked that if all stream replication methods are the same as the expected one at line 98. While we checked that if we have a replication key for any stream then it must be `INCREMENTAL` stream at line 103.

Comment (karanpanchal-crest) [reply tests/test_facebook_discovery.py:101]: But isn't that already being checked at line 98?
At line 98,
Stream A -> Expected replication method = "Incremental", Actual replication method = "Incremental" ===> True
Stream B -> Expected replication method = "FULL_TABLE", Actual replication method = "FULL_TABLE" ===> True

At line 103, you are checking the same thing, but more specifically
Stream A -> If replication key -> Actual replication method == "Incremental"  ==> True
Stream B -> Not verifying anything here.

In a nutshell, I believe check at line 103 is not required. Let me know if you think otherwise

Comment (savan-chovatiya) [reply tests/test_facebook_discovery.py:101]: These checks are part of standard tests defined by Stitch and also these are already there in a test, we only moved it out from the if condition as we have fixed this issue for some of the streams.

**Review by KrisPersonal — APPROVED** [review]

File: tests/test_facebook_discovery.py (hunk 3 above)
Comment (kspeer825) [reply tests/test_facebook_discovery.py:101]: Savan is correct. This assertion will be redundant only if expectations are correct. It's real value is to catch invalid expectations set by a tester.

**Review by kspeer825 — APPROVED** [review]
